src/merkle/store/tests.rs

```diff
@@ -15,7 +15,8 @@ const EMPTY: Word = [ZERO; WORD_SIZE];
 #[test]
 fn test_root_not_in_store() -> Result<(), MerkleError> {
     let mtree = MerkleTree::new(LEAVES4.to_vec())?;
-    let store = MerkleStore::default().with_merkle_tree(LEAVES4)?;
+    let mut store = MerkleStore::new();
+    store.extend(mtree.inner_nodes());
```

Comment:
Should we change the approach to make this work as follows?:
```Rust
let mut store = MerkleStore::new();
store.extend(&mtree);
```Rust
Also, I'm wonder if we can make `From` work as well - maybe something like:
```
let store = MerkleStore::from(&mtree);`
```

Reply:
I think to support this we might have to implement iterators on `MerkleTree`, `SimpleSmt` etc. Once we land on how to do this, let's also revisit #122 to follow the same approach.

Reply:
Added `From` conversion for `MerkleTree`, `SimpleSMT`, and `Mmr`. I used the `inner_nodes` iterator for that.